test/inductor/test_torchinductor.py

```diff
@@ -10358,6 +10358,19 @@ def fn(x):
         x = torch.rand([4, 4, 3], dtype=torch.float64)
         self.common(fn, (x,))
 
+    @skipCUDAIf(not SM80OrLater, "uses bfloat16 which requires SM >= 80")
+    def test_dtypeview(self):
+        # https://github.com/pytorch/pytorch/issues/126338
+        @torch.compile
+        def fn(x, y):
+            x = x.view(torch.float16)
+            y = y.view(torch.float16) + 1
+            return x @ y
+
+        x = torch.randn((2, 2), device=self.device, dtype=torch.bfloat16)
+        y = torch.randn((2, 2), device=self.device, dtype=torch.bfloat16)
+        fn(x, y)
+
```

Comment:
curious: would it be easy to also add unit test for the original `functional_collective -> view.dtype -> wait` use case? thanks!

Reply:
I think this example https://github.com/pytorch/pytorch/issues/126338#issuecomment-2113608423 is the best mini-repro for testing.

Reply:
You don't need fp8 to do the repro. and you can use multiple gpus, see `TEST_MULTIGPU`.

Reply:
we can use fake ProcessGroup too, this might be a good example: https://github.com/pytorch/pytorch/blob/7bda23ef8495a22bde8b7861f5d4f3f015cee483/test/distributed/test_c10d_functional_native.py#L595, we can add the collective unit test directly to that file and leverage the test infra there.

Reply:
@eellison @yf225 Thanks for your information! I found it is easier to add a test in `test_c10d_functional_native.py`. did it in https://github.com/pytorch/pytorch/pull/128883/commits/331c865f6a6f5b5eb3de6083c67ba5b6ee2ecbec